index.js

```diff
@@ -123,18 +127,16 @@ function connect (appKey, accessToken, _opts) {
     next()
   }
 
-  function end (done) {
+  function end(done) {
     ended = true
-
-    if (request) {
-      request.end()
-      request.on('end', done)
-    } else {
-      done()
-    }
+    if (request) request.destroy()
+    if (responseStream) responseStream.destroy()
+    done()
   }
 
-  function emitError (err) {
+  function emitError(err) {
+    // if the stream is ending, we can just swallow any errors
```

Comment:
this is maybe controversial, but my thinking is that if they call to end the stream they don't care about any errors after that; otherwise they might get weird errors from `zlib` due to attempts to decode incomplete buffers etc.

Reply:
a different fix would be to continue emitting the errors, but document the potential behavior; leave it up to the caller to remove their error handlers before ending the stream. how weird of a sentence that is _probably_ highlights that your fix is right: if we're like "oh, remove your error handlers" that means that we're saying you probably don't care about the errors, which means we should probably not make anyone deal with that.

given this is a library that no one should care about the internal implementation details, it seems weird to make people care about that behavior rather than just handling it. the only exception would be if we perform any "dangerous" action, like consuming something from the RTDS stream that would be otherwise lost if they weren't made aware of this error. given we deal in offsets, i'm almost certain this isn't true so I think your fix is right.

i'm just putting the above sentence here in case you read it and disagree; i think this is fine.